app/assets/javascripts/discourse/models/category.js.es6

```diff
@@ -243,7 +243,11 @@ Category.reopenClass({
   },
 
   findSingleBySlug(slug) {
-    return Category.list().find(c => Category.slugFor(c) === slug);
+    if (Discourse.SiteSettings.slug_generation_method !== "encoded") {
```

Comment:
Could you use `this.siteSettings`? Should be available in the Category model.

Reply:
I could use `Category.list()[0].siteSettings.slug_generation_method` to access it but IMO is ugly as hell, and looks like almost every model is using the global:

![image](https://user-images.githubusercontent.com/1385470/66859600-fa32a500-ef61-11e9-8456-adf534c5e506.png)

Reply:
Ugh, good point. We should make it a service eventually.